src/seedu/addressbook/data/person/Name.java

```diff
@@ -22,11 +22,29 @@
      * @throws IllegalValueException if given name string is invalid.
      */
     public Name(String name) throws IllegalValueException {
-        String trimmedName = name.trim();
-        if (!isValidName(trimmedName)) {
+        String capitalizedName = capitalizeFirstCharForEveryWord(name.trim());
+        if (!isValidName(capitalizedName)) {
             throw new IllegalValueException(MESSAGE_NAME_CONSTRAINTS);
         }
-        this.fullName = trimmedName;
+        this.fullName = capitalizedName;
+    }
+
+    /**
+     * Format name before adding into address book
+     * (Capitalize first character of every word in name & the rest in lower case)
+     *
+     * @param name string argument
+     * @return formatted name argument
+     */
+    private static String capitalizeFirstCharForEveryWord(String name) {
+        String[] partOfName = name.split("\\s+");
+        String fullName = "";
+        for(String word : partOfName){
```

Comment:
Just a minor comment. Java reserved words such as `for` should always be followed by a white space. You may refer to the Layout section in our [coding standards](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html).